src/clj/firestore_clj/google.clj

```diff
@@ -0,0 +1,72 @@
+(ns firestore-clj.google
+  (:require
+   [firestore-clj.google.fake-credentials :as fake-credentials])
+  (:import
+   (clj.firebase.google
+    EmulatorChannelConfigurator
+    FakeCredentials)
+   (com.google.api.gax.grpc InstantiatingGrpcChannelProvider)
+   (com.google.api.gax.rpc FixedHeaderProvider)
+   (com.google.api.core ApiFunction)
+   (com.google.auth
+    Credentials)
+   (com.google.auth.oauth2
+    GoogleCredentials)
+   (com.google.cloud.firestore
+    Firestore
+    FirestoreOptions
+    FirestoreOptions$Builder)))
+
+(defn mk-fake-credentials
+  []
+  ^Credentials
+  (FakeCredentials.))
+
+(defn mk-fixed-header-provider
+  []
+  (FixedHeaderProvider/create {"Authorization" "Bearer owner"}))
+
+(defn mk-emulator-channel-configurator
+  []
+  ^ApiFunction
+  (EmulatorChannelConfigurator.))
+
+(defn mk-emulator-grpc-channel-provider
+  ^InstantiatingGrpcChannelProvider
+  [emulator-host]
+  (let [p (-> (InstantiatingGrpcChannelProvider/newBuilder)
+              (.setEndpoint emulator-host)
+              (.setChannelConfigurator (mk-emulator-channel-configurator))
+              (.setHeaderProvider (mk-fixed-header-provider))
+              (.build))]
+    p))
+
+(defn emulator-client
+  "Gets a client i.e. using the emulator"
+  ^Firestore
+  [project-id emulator-host]
+  (let [credentials (mk-fake-credentials)
+        channel-provider (mk-emulator-grpc-channel-provider emulator-host)
+        ^FirestoreOptions options (->
+                                   (FirestoreOptions/getDefaultInstance)
+                                   (.toBuilder)
+                                   ^FirestoreOptions$Builder
+                                   (.setProjectId project-id)
+                                   (.setChannelProvider channel-provider)
+                                   ^FirestoreOptions$Builder
+                                   (.setCredentials credentials)
```

Comment:
As far as I can tell the fake credentials are not used, except to guarantee validation works for the builder.

The real authorization happens using the `fixed-header-provider`

https://github.com/polvotech/firestore-clj/pull/4/files#diff-8dcfb082370a13ada9fe204cec4f7a1aR27